start.py

```
@@ -450,32 +450,57 @@ def create_directories() -> bool:
+        print_success("React production build is current")
+        return True
+
+    npm = shutil.which("npm")
+    if not npm:
+        print_error("npm is required because frontend/dist is missing or stale")
+        return False
+
+    try:
+        if not (frontend_path / "node_modules").is_dir():
+            print_info("Installing frontend dependencies with npm ci")
+            subprocess.run([npm, "ci"], cwd=frontend_path, check=True, timeout=600)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)